src/Catty/DataModel/UserData/UserVariable.swift

```diff
@@ -89,4 +89,12 @@
         }
         variableLocker.unlock()
     }
+
+    // Truncate to prevent MTLTextureDescriptor maximum allowed size error
+    func truncateIfLengthExceeded(value: String?) -> String? {
+        if let value = value, value.count > 160 {
+            return value[..<String.Index(utf16Offset: 160, in: value)] + "..."
```

Comment:
Please add a constant for the value 160 in `SpriteKitDefines` (with a proper name).